test/Chainweb/Test/CutDB.hs

```diff
@@ -398,23 +399,15 @@ tryMineForChain miner webPact cutDb c cid = do
     x <- testMineWithPayloadHash (Nonce 0) t payloadHash cid c
     case x of
         Right (T2 h c') -> do
-            validate h outputs
             addCutHashes cutDb (cutToCutHashes Nothing c')
+                { _cutHashesHeaders = HM.singleton (_blockHash h) h
+                , _cutHashesPayloads = HM.singleton (_blockPayloadHash h) (payloadWithOutputsToPayloadData outputs)
+                }
             return $ Right (c', cid, outputs)
         Left e -> return $ Left e
   where
     parent = ParentHeader $ c ^?! ixg cid -- parent to mine on
 
-    payloadDb = view cutDbPayloadCas cutDb
-    webDb = view cutDbWebBlockHeaderDb cutDb
-    pact = _webPactExecutionService webPact
-
-    validate h outputs = do
-        let pd = payloadWithOutputsToPayloadData outputs
-        void $ _pactValidateBlock pact h pd
-        addNewPayload payloadDb outputs
-        insertWebBlockHeaderDb webDb h
-
```

Comment:
This code was hiding a bug in `fakePact`. Headers and Payloads should only be added to the chain database via `addCutHashes`.

Reply:
By restricting `view cutDbPayloadCas cutDb` to `HasCasLookup` (and not providing write access via `IsCas` as before), it's not possible any more to bypass the cut validation pipeline.